### PR TITLE
Add roadmap

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: Roadmap.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+1. Mustache to handlebars
+
+    Handlebars keeps most compatibility with Mustache template syntax, and meanwhile it is more powerful. It supports partials with parameters, which makes componentize possible. It also contains [Built-In Helpers](http://handlebarsjs.com/#builtins) such as `if` conditional and `each` iterator.
+
+2. Support new syntax
+    1. React  
+[React](https://facebook.github.io/react/) is popularly used when developing web applications. If we support `JSX` and leverage `React.Component`, is it more convenient for front-end developers to integrate with docfx?
+
+    2. Razor  
+[RazorEngine](https://antaris.github.io/RazorEngine/) as the template engine for ASP.NET, is more friendly to C# developers. It also support partials with parameters.
```
Comment (superyyrrzz) [inline Roadmap.md]: >more friendly [](start = 93, length = 13)

friendlier

Comment (superyyrrzz) [inline Roadmap.md]: >support [](start = 133, length = 7)

supports

File: Roadmap.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+* Highlighted clickable method declaration, e.g. *[String]() ToString([int]() a)*
+* Cross platform support
+    * Dotnet-core migration
+    * Docker
+* VSCode extension
+    * TOC Preview
+    * Intellisense for docfx.json, toc.yml, and DFM syntax
+* Localization and versioning support
+* More attractive themes
+* Sandcastle advanced features
```
Comment (superyyrrzz) [inline Roadmap.md]: * Support programming languages other than C#, VB, like Python, JavaScript...

File: Roadmap.md (hunk 1 above)
Comment (vicancy) [inline Roadmap.md]: both are OK https://www.reddit.com/r/grammar/comments/354wta/more_friendly_vs_friendlier_comparativesuperlative/

**Comment by superyyrrzz** [comment]
:shipit:

File: Roadmap.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+- [x] Feature already available in [docfx](RELEASENOTE.md)
+
+## Near-term
+
+### Schema based document processor
+
+Currently every new language introduces in a new data model, and results in a new plugin and a new template. With schema based document processor, what needs for a new language is simply a schema file.
+
+This schema file can be possibly leveraged in Localization scenario, to define which property is localizable, etc.
```
Comment (hellosnow) [inline Roadmap.md]: >Localization [](start = 46, length = 12)

`Localization`?

File: Roadmap.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+Looks like *logic* is important, and it is more convenient to allow using *logic* when writing template. With the popularity of [React](https://facebook.github.io/react/) and [Vue.js](http://cn.vuejs.org/), *declarative views* becomes intriguing with its predicatability and ease of use.
+
+2. Component based templates
+We can consider **partials** as **components** when **partials** accepts parameters. For example, with `> inheritance.partial uid='System.String'`, we can move the *expand* model logic from C# code to templates.
+
+#### Proposals:
+1. Mustache to handlebars
+
+    Handlebars keeps most compatibility with Mustache template syntax, and meanwhile it is more powerful. It supports partials with parameters, which makes componentize possible. It also contains [Built-In Helpers](http://handlebarsjs.com/#builtins) such as `if` conditional and `each` iterator.
```
Comment (hellosnow) [inline Roadmap.md]: >componentize [](start = 156, length = 12)

componentization?

File: Roadmap.md (hunk 3 above)
Comment (vicancy) [inline Roadmap.md]: `localizable` means `able to be localized`

**Comment by hellosnow** [comment]
:shipit:

File: Roadmap.md (hunk 3 above)
Comment (vwxyzh) [inline Roadmap.md]: And overwrite behavior?

**Pull request merged**